packages/api/src/bsky-agent.ts

```diff
@@ -289,6 +307,18 @@ export class BskyAgent extends AtpAgent {
         if (pref.birthDate) {
           prefs.birthDate = new Date(pref.birthDate)
         }
+      } else if (
+        AppBskyActorDefs.isFeedViewPref(pref) &&
+        AppBskyActorDefs.validateFeedViewPref(pref).success
+      ) {
+        const { $type, feed, ...v } = pref
+        prefs.feedViewPrefs[pref.feed] = { ...FEED_VIEW_PREF_DEFAULTS, ...v }
+      } else if (
+        AppBskyActorDefs.isThreadViewPref(pref) &&
+        AppBskyActorDefs.validateThreadViewPref(pref).success
+      ) {
+        const { $type, ...v } = pref
```

Comment:
could do this `$type` removal for `feedViewPrefs` values also (couple lines above), or not bother

Reply:
wdym?

Reply:
I could have sworn that `const { $type, feed, ...v } = pref` on line 314 was not there when I reviewed, but I see it now!